cachi2/core/package_managers/pip.py

```diff
@@ -1477,37 +1481,32 @@ def _process_req(
 ) -> dict[str, Any]:
     download_info["kind"] = req.kind
     download_info["requirement_file"] = str(requirements_file.file_path.subpath_from_root)
-    download_info["hash_verified"] = False
+    download_info["missing_checksums"] = True
+    # "package_type" is *only* needed for PyPI deps
+    download_info["package_type"] = ""
 
-    def _hash_verify(path: Path, checksum_info: Iterable[ChecksumInfo]) -> bool:
-        verified: bool = False
+    def _checksum_match(path: Path, checksum_info: Iterable[ChecksumInfo]) -> None:
```

Comment:
It looks like this function does not only verify a checksum, but also produces a side-effect: it unlinks a Path. Would something like `_unlink_path_if_no_match_found`  be a more descriptive name?

Reply:
Good point